server/manager/src/main/java/org/apache/accumulo/manager/TabletGroupWatcher.java

```diff
@@ -640,10 +645,12 @@ public void run() {
     boolean lookForTabletsNeedingVolReplacement = true;
 
     while (manager.stillManager()) {
-      // slow things down a little, otherwise we spam the logs when there are many wake-up events
-      sleepUninterruptibly(100, TimeUnit.MILLISECONDS);
-      // ELASTICITY_TODO above sleep in the case when not doing a full scan to make manager more
-      // responsive
+      if (!eventHandler.isNeedsFullScan()) {
+        // If an event handled by the EventHandler.RangeProcessor indicated
+        // that we need to do a full scan, then do it. Otherwise wait a bit
+        // before re-checking the tablets.
+        sleepUninterruptibly(100, TimeUnit.MILLISECONDS);
+      }
```

Comment:
I am wondering if we should just remove this sleep for now and add it back if needed.  This sleep is coming from pre elasticity code and so much as changed in elasticity that it may not be needed or maybe something different is needed, like maybe we will need a configurable minimum time between full scans.

This sleep existed when every event could cause a full scan prior to elasticity.  Now in elasticity there are event types and not all types cause a full scan.

```suggestion
```

Reply:
Thinking about it a bit more, not really sure if it should be removed or not.  The sleep that existed prior to elasticity feels like a hacky change to address a very specific problem the older code was experiencing.  Not sure if that problems is still relevant.  The changes in this PR are fine, but the existing foundation they build on seems shaky.  Things like this need to be refined in testing of elasticity.